test/appium/views/profile_view.py

```diff
@@ -96,29 +95,35 @@ class ShareMyContactKeyButton(BaseButton):
 
     def __init__(self, driver):
         super(ShareMyContactKeyButton, self).__init__(driver)
-        self.locator = self.Locator.xpath_selector('//*[@text="SHARE MY CONTACT CODE"]')
+        self.locator = self.Locator.accessibility_id('share-my-contact-code-button')
 
 
 class EditButton(BaseButton):
 
     def __init__(self, driver):
         super(EditButton, self).__init__(driver)
-        self.locator = self.Locator.xpath_selector('//*[@text="EDIT"]')
+        self.locator = self.Locator.accessibility_id('edit-button')
 
 
 class ConfirmButton(BaseButton):
 
     def __init__(self, driver):
         super(ConfirmButton, self).__init__(driver)
-        self.locator = self.Locator.xpath_selector('(//android.view.ViewGroup[@content-desc="icon"])[1]')
+        self.locator = self.Locator.accessibility_id('done-button')
 
 
 class CrossIcon(BaseButton):
 
     def __init__(self, driver):
         super(CrossIcon, self).__init__(driver)
-        self.locator = self.Locator.xpath_selector(
-            '(// android.view.ViewGroup[@ content-desc="icon"])[1]/android.view.View')
+        self.locator = self.Locator.accessibility_id('done-button')
+
+
+class AdvancedButton(BaseButton):
+
+    def __init__(self, driver):
+        super(AdvancedButton, self).__init__(driver)
+        self.locator = self.Locator.xpath_selector('//*[@text="Advanced"]')
```

Comment:
Hm, do we need accessibility label for it in order to run the tests on both Android and iOS? If not, maybe we can write it down and add it later.

Reply:
Yes, we definitely need it